inventory/packages/wua_windows.go

```diff
@@ -15,15 +15,139 @@ package packages
 
 import (
 	"fmt"
+	"sync"
 
 	ole "github.com/go-ole/go-ole"
 	"github.com/go-ole/go-ole/oleutil"
 )
 
-type (
-	IUpdateSession    = *ole.IDispatch
-	IUpdateCollection = *ole.IDispatch
-)
+var wuaSession sync.Mutex
+
+// IUpdateSession is a an IUpdateSession.
+type IUpdateSession struct {
+	obj *ole.IUnknown
+	ses *ole.IDispatch
+}
+
+func NewUpdateSession() (*IUpdateSession, error) {
+	wuaSession.Lock()
+	if err := ole.CoInitializeEx(0, ole.COINIT_MULTITHREADED); err != nil {
+		return nil, err
```

Comment:
If `NewUpdateSession` returns an error, will `wuaSession` ever be unlocked? It doesn't look like that will happen.

Reply:
fixed

Reply:
Also FYI this library still has some sharp edges that you need to be aware of to use, I tried to make it simpler to understand but the effort to try to make this really universally safe to use is not really worth it as it is meant for the osconfig agent and we manage our action serialization well in there.